### PR TITLE
refactor(deployment): GCP marketplace detailed deployer KSA's permissions

### PR DESCRIPTION
**Pull request by rmgogogo**
**Description of your changes:**
feat(deployment): Update deployer ksa roles so that it can unlock MKP side change which refines deployer KSA's permissions.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by rmgogogo** [comment]
it's for b/159616919 internal tracking.

**Comment by rmgogogo** [comment]
I'm also OK we don't cherrypick but release it in next cut.

**Comment by Bobgy** [comment]
Updated title to `refactor` because this is an internal change, users don't need to read it from release note.

**Comment by Bobgy** [comment]
@rmgogogo next cut will be 1 month from now, looking at the internal issue, we should get this public ASAP.
So prefer cherry picking it to 1.0 release.

File: manifests/gcp_marketplace/schema.yaml [hunk 1]
```diff
@@ -92,6 +92,40 @@ x-google-marketplace:
...
+    # - ml-pipeline
+    # - ml-pipeline-persistenceagent-role
+    # - ml-pipeline-scheduledworkflow-role
+    # - ml-pipeline-viewer-controller-role
+    # - ml-pipeline-ui
+    # - pipeline-runner
+    - type: Role
+      rulesType: CUSTOM
+      rules:
+      - apiGroups: ['argoproj.io']
```
Comment (Bobgy) [inline manifests/gcp_marketplace/schema.yaml:107]: I didn't get it, why does deployer need these permissions?

Deployer doesn't deploy these CRs directly.

Comment (rmgogogo) [reply manifests/gcp_marketplace/schema.yaml:107]: I also didn't get it, Erika suggested the change.

**Comment by rmgogogo** [comment]
> @rmgogogo next cut will be 1 month later, looking at the internal issue, we should get this public ASAP.
> So prefer cherry picking it to 1.0 release.

I'm OK. We may sync with Erika to let her postpone her side change as well. I think it should be fine.

**Comment by Bobgy** [comment]
Moving to post 1.0

**Comment by Bobgy** [comment]
Confirmed, we only need to add resources the deployer is going to deploy.

**Comment by Bobgy** [comment]
MKP test was already failing in postsubmit, I'll need to fix that first.

**Comment by Bobgy** [comment]
The errors I found in logs:

> error when creating "/data/resources.yaml": roles.rbac.authorization.k8s.io "proxy-agent-runner" is forbidden: user "system:serviceaccount:apptest-117g7fqa:apptest-117g7fqa-deployer-sa" (groups=["system:serviceaccounts" "system:serviceaccounts:apptest-117g7fqa" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:

It seems the deployer needs to have all permissions any role we deploy need to have.

**Pull request merged**